design/src/aws_config.md

```diff
@@ -0,0 +1,92 @@
+# AWS Configuration RFC
+
+An AWS SDK loads configuration from many different location. Some of these locations can be loaded synchronously. Some
+are async. Others may actually use AWS services.
+
+This document proposes an overhaul to the Config design to facilitate three things:
+
+1. Future-proof: It should be easy to add additional sources of region and credentials, sync and async, from many
+   sources, including code generated services.
+2. Ergonomic: There should be one obvious way to create an AWS service client.
+3. Shareable: A config object should be usable to configure multiple AWS services.
+
+## New customer experience
+
+In Cargo.toml:
+
+```toml
+[dependencies]
+aws-sdk-dynamo = "0.1"
+aws-config = "0.5"
+```
+
+```rust
+// pub use of aws_types::SharedConfig
+use aws_config::SharedConfig;
+async fn main() {
+   // loading a shared config may make network calls, etc.
+   let config = SharedConfig::load_from_environment().await;
```

Comment:
If a customer wants to load from environment, but also change the sleep implementation, what would this look like?

Reply:
`SharedConfig::builder().sleep(...).build().await`